logical/plugin/plugin.go

```diff
@@ -18,6 +18,13 @@ import (
 // Register these types since we have to serialize and de-serialize tls.ConnectionState
 // over the wire as part of logical.Request.Connection.
 func init() {
+	// Common basic structs
+	gob.Register([]interface{}{})
+	gob.Register(map[string]interface{}{})
```

Comment:
Even though these structs are not used right now, I've added it to guard against potential use in the future.

Reply:
I wonder if these are already registered, I think go registers a lot of builtin types off the bat